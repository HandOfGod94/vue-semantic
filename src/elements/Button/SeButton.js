import Vue from 'vue'
import Component from 'vue-class-component'

@Component({
  name: 'se-button',
  props: {
    color: String,
    label: String,
    icon: String,
    basic: Boolean,
    inverted: Boolean,
    disabled: Boolean,
    loading: Boolean,
    size: String,
    float: String,
    compact: Boolean,
    fluid: Boolean,
    circular: Boolean
  }
})
class SeButton extends Vue {
  render (h) {
    return (
      <button class={this.classes}>
        { this.icon && <i class={`${this.icon} icon`}></i> }
        {this.label}
        {this.$slots.default}
      </button>
    )
  }

  get classes () {
    const classes = {
      [this.size]: this.size,
      compact: this.compact,
      circular: this.circular,

      ui: true,

      [`${this.float} floated`]: this.float,
      disabled: this.disabled,
      inverted: this.inverted,
      [this.color]: this.color,
      loading: this.loading,
      labeled: this.label,
      icon: this.icon,
      basic: this.basic,

      button: true
    }
    return classes
  }
}

export default SeButton
