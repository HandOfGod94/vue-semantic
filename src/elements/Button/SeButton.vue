<template>
  <component :is="as" :class="classObject">
    <i :class="`${this.icon} icon`" v-if="classObject.icon"></i>
    <slot name="label">
      {{label}}
    </slot>
    <slot></slot>
  </component>
</template>

<script>
export default {
  name: 'se-button',
  props: {
    as: { type: String, default: 'button' },
    active: Boolean,
    attach: String,
    color: String,
    label: String,
    icon: String,
    basic: Boolean,
    inverted: Boolean,
    disabled: Boolean,
    loading: Boolean,
    size: String,
    float: String,
    compact: Boolean,
    fluid: Boolean,
    circular: Boolean
  },
  computed: {
    classObject () {
      return {
        [this.size]: this.size,
        compact: this.compact,
        circular: this.circular,
        ui: true, // Semantic ui class
        [`${this.attach} attached`]: this.attach,
        [`${this.float} floated`]: this.float,
        disabled: this.disabled,
        inverted: this.inverted,
        [this.color]: this.color,
        loading: this.loading,
        labeled: this.label,
        icon: this.icon,
        basic: this.basic,
        button: true
      }
    }
  }
}
</script>
