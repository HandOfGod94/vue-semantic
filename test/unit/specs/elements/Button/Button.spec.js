import Vue from 'vue'
import SeButton from '@/elements/Button/SeButton'

describe('SeButton Color', () => {
  it('should have correct classes for primary color', () => {
    const propsData = { propsData: { color: 'primary' } }
    const Constructor = Vue.extend(SeButton)
    const vm = new Constructor(propsData).$mount()
    expect(vm.$el.classList.contains('primary')).toBeTruthy()
    expect(vm.$el).toMatchSnapshot()
  })

  it('should have correct classes for secondary color', () => {
    const propsData = { propsData: { color: 'secondary' } }
    const Constructor = Vue.extend(SeButton)
    const vm = new Constructor(propsData).$mount()
    expect(vm.$el.classList.contains('secondary')).toBeTruthy()
    expect(vm.$el).toMatchSnapshot()
  })

  it('should have correct classes for custom color', () => {
    const propsData = { propsData: { color: 'orange' } }
    const Constructor = Vue.extend(SeButton)
    const vm = new Constructor(propsData).$mount()
    expect(vm.$el.classList.contains('orange')).toBeTruthy()
    expect(vm.$el).toMatchSnapshot()
  })
})
