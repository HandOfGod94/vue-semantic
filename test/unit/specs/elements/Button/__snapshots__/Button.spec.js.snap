// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeButton Color should have correct classes for primary color 1`] = `
<button
  class="ui primary button"
/>
`;
